### PR TITLE
tbg pathlist

### PR DESCRIPTION
**Pull request by Anton-Le**
Modification of the TBG script which would implement #3189.

### CODE REVIEW

File: bin/tbg [hunk 1]
```diff
@@ -272,7 +272,15 @@ help()
 }
 
 #!/usr/bin/env bash
-initCall="$0 $*"
+#Check whether the provided path is relative or absolute and
+#convert it to an abolute path
+script="$0"
+case $script in
+        /*) absScriptPath=$script;; #called using abs. path
+        ./*) absScriptPath=$PWD/${script#"./"};; #called using ./, remove it
```
Comment (psychocoderHPC) [inline bin/tbg]: There is no need for two three cases see sugestion: https://github.com/ComputationalRadiationPhysics/picongpu/issues/3189#issuecomment-591968591

File: include/picongpu/particles/ParticlesInit.kernel [hunk 2]
```diff
@@ -42,7 +42,7 @@ namespace picongpu
      *
      *  That means:
      *     The REAL number of particles density in units of volume**-3,
-     *       normed to UNIT_LENGHT**3
```
Comment (psychocoderHPC) [inline include/picongpu/particles/ParticlesInit.kernel]: please move changes from this file into a separate PR, even if it is doubling the work. The reason is that bose changes are independent from each other.

**Comment by psychocoderHPC** [comment]
please squash your commits. Please have also a look to your local git config. The last commit is different to the first commits.

File: include/picongpu/particles/ParticlesInit.kernel [hunk 3]
```diff
@@ -64,6 +64,10 @@ namespace picongpu
         );
 
         float_X const densityRatioOfSpecies = traits::GetDensityRatio< T_Species >::type::getValue( );
+        PMACC_CASSERT_MSG(
+            Please_set_the_densityRatio_to_a_value_greater_than_0,
+            densityRatioOfSpecies > float_X( 0.0 )
```
Comment (psychocoderHPC) [inline include/picongpu/particles/ParticlesInit.kernel]: density ratio is not `constexpr` and can therefore not checked at compile time. We also have a PR #3109  where we started to introduce a check for `MassRatioValue` but we discarded it. We like to move more stuff into runtime (if possible).

File: bin/tbg (hunk 1 above)
Comment (Anton-Le) [reply bin/tbg]: I don't think so. Consider the minimal working example:

    #!/bin/bash
    echo "$0"
    curpath=$0
    #convert to abs path
    case $curpath in
            /*) abspath=$curpath;;
           ./*) abspath=$PWD/${curpath#"./"};;
            *) abspath=$PWD/$curpath;;
    esac
    echo "Absolute path: "$abspath

Save it as a script as `realpathExample.sh` in a directory that is NOT in your `PATH`. Then using the suggestion from #3189 would result in an error and no absolute path. 

Removing the default case `*)` from the above will, for instance, prevent the user from obtaining the absolute path when running using `bash realpathExample.sh`.

**Comment by psychocoderHPC** [comment]
@Anton-Le please squash the commits and remove all not required changes that I can review the PR.

**Comment by psychocoderHPC** [comment]
@Anton-Le ping, do you forgot this PR?

**Comment by psychocoderHPC** [comment]
ping

**Comment by psychocoderHPC** [comment]
@Anton-Le PING PING

File: include/picongpu/particles/ParticlesInit.kernel (hunk 3 above)
Comment (Anton-Le) [reply include/picongpu/particles/ParticlesInit.kernel]: Removed from the commit

File: bin/tbg [hunk 4]
```diff
@@ -272,7 +272,15 @@ help()
 }
 
 #!/usr/bin/env bash
-initCall="$0 $*"
+#Check whether the provided path is relative or absolute and
+#convert it to an abolute path
+script=$0
+case $script in
+        /*) absScriptPath=$script;; #called using abs. path
+        ./*) absScriptPath=$PWD/${script#"./"};; #called using ./, remove it
```
Comment (PrometheusPi) [inline bin/tbg:280]: This could be simplified by using `realpath`. It is installed on most systems (we use).

Comment (Anton-Le) [reply bin/tbg]: Admittedly, this is a much more elegant solution which should work assuming the system is running a linux distribution younger than ~2012!

I have incorporated it into the script.
